Dev

FireInstall opened the pull request:
The newish Materials where missing again, and I was annoyed.

FireInstall committed:
removed doubled PluginPrefix in Logs
Fixed surrounding quotation marks of messages
Removed unused "defaultMessages"

FireInstall committed:
Better Message handling (thanks Feuerreiter once again for Padlock!)
some code messiness fixed

FireInstall committed:
Fix white/blacklist not using correct categories

FireInstall committed:
* Fix region checking code. It was picking up all regions in all worlds
   at specified coordinates. Now only regions applicable to the current world
   at the specified location are checked.
 * Treat all air material types equivalently.
 Commit taken from:
 totemo/RFChairs author: totemo

FireInstall committed:
proper relocate bstats

FireInstall committed:
some clean up

FireInstall committed:
load legacy named whitelist

FireInstall committed:
Fix flying players if block was broken

FireInstall committed:
forgot to change the description

FireInstall committed:
added tabCompletion

FireInstall commented:
![](https://github.trello.services/images/mini-trello-icon.png) [RifleChairs: Fixes](https://trello.com/c/Fb2S9pPp/258-riflechairs-fixes)

FireInstall committed:
better logging
paper plugin
1.21

seema84 commented:
Can you perhaps also fix the incorrect seat height on slap? The adjustments in the config have no effect (only for slaps).
Is perhaps only "g" which is too much for slab-seating-position"g": 0.5,0.3,0.5

FireInstall commented:
You are completely correct, good catch, thanks!
I have a fix locally, will push as soon as I can verify against our config

FireInstall committed:
update gradle and its plugins
update dependencies
update minecraft to the latest stable paper build (1.21.4 does not work currently, probably do to the hard fork restructure)
fix config key "slab-seating-position" having a tailing "g", thanks @seema84
moved the default values up, as the old ones looked wrong and resulted in glitching through blocks in case of carpets (some minecraft update changed the point of seating on armor stands down)
removed helper method to check if a Material is air, as Material already has a method for this.

FireInstall committed:
Added download plugins to runServer task, to test with worldguard without having to download it manually.
Removed wrong comment

FireInstall committed:
remove legacy "allowed-chairs" from config to not confuse users with old configs

seema84 commented:
> You are completely correct, good catch, thanks! I have a fix locally, will push as soon as I can verify against our config

thanks <3

Schpammer commented on src/main/java/com/rifledluffy/chairs/dependencymanagers/WorldGuardManager.java:
```diff
@@ -0,0 +1,76 @@
+        }
+    }
+
+    public @Nullable WorldGuardPlugin getWorldGuard() { // todo cache the plugin
```
Isn't the todo just to create a class variable?

FireInstall commented:
You are right. Don't know why I just added this comment instead of changing it right away.
Probably chased another thought

FireInstall committed:
use ComponentLogger
Drop deprecated UTF8BundleControl
bump gradle

Didn't update to newer  minecraft version to wait on PR #3

joshi1999 commented on src/main/java/com/rifledluffy/chairs/config/ConfigManager.java:
```diff
@@ -93,15 +65,11 @@ public void saveConfig() {
             config.save(configFile);
         } catch (IOException e) {
-            plugin.getLogger().info("[Rifle's Chairs] Could not save config.yml!");
+            plugin.getComponentLogger().info("[Rifle's Chairs] Could not save config.yml!");
```
Inconsistent, "[Rifle's Chars]" is not used in the other error messages; Also the Logger will handle this.

FireInstall committed:
Fix double plugin prefix in logger

FireInstall commented:
good catch, thanks!
